tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Testing/MsTestV2/MsTestIntegration.cs

```diff
@@ -103,6 +81,44 @@ internal static Test OnMethodBegin<TTestMethod>(TTestMethod testMethodInstance,
         return test;
     }
 
+    internal static void UpdateTestParameters<TTestMethod>(Test test, TTestMethod testMethodInstance, string displayName = null)
+        where TTestMethod : ITestMethod
+    {
+        var testMethod = testMethodInstance.MethodInfo;
+        var testMethodArguments = testMethodInstance.Arguments;
+        var testName = testMethodInstance.TestMethodName;
+
+        // Get test parameters
+        var methodParameters = testMethod.GetParameters();
+        if (methodParameters?.Length > 0)
+        {
+            var testParameters = new TestParameters
+            {
+                Metadata = new Dictionary<string, object>(),
+                Arguments = new Dictionary<string, object>()
+            };
+
+            if (!string.IsNullOrEmpty(displayName) && displayName != testName)
+            {
+                testParameters.Metadata[TestTags.MetadataTestName] = displayName;
+            }
+
+            for (var i = 0; i < methodParameters.Length; i++)
+            {
+                if (testMethodArguments != null && i < testMethodArguments.Length)
+                {
+                    testParameters.Arguments[methodParameters[i].Name ?? $"{i}"] = Common.GetParametersValueData(testMethodArguments[i]);
```

Comment:
I realise this existing before, but we probably want culture invariant to string right?
```suggestion
                    testParameters.Arguments[methodParameters[i].Name ?? i.ToString(CultureInfo.InvariantCulture)] = Common.GetParametersValueData(testMethodArguments[i]);
```

Reply:
done in [f4d4b7a](https://github.com/DataDog/dd-trace-dotnet/pull/5381/commits/f4d4b7acfdfab96cfbb9680c8ef1299db40978f7)